klee-replay does not work without the full path to the executable

When I use trying the Coreutils according to the https://klee.github.io/tutorials/testing-coreutils/ 
I have the problem,like "KLEE: WARNING ONCE: Alignment of memory from call "malloc" is not modelled. Using alignment of 8." But there are uclibc and posix. And after warnings, the output contains some blank lines and ends with 'KLEE: done'.  

I use Ubuntu18 and the latest version Coreutils.

~/coreutils/obj-llvm/src$ klee --libc=uclibc --posix-runtime ./echo.bc --sym-arg 3

KLEE: NOTE: Using POSIX model: /usr/local/lib/klee/runtime/libkleeRuntimePOSIX.bca
KLEE: NOTE: Using klee-uclibc : /usr/local/lib/klee/runtime/klee-uclibc.bca
KLEE: output directory is "/home/wei/coreutils/obj-llvm/src/./klee-out-15"
KLEE: Using STP solver backend
KLEE: WARNING: undefined reference to function: __ctype_get_mb_cur_max
KLEE: WARNING: executable has module level assembly (ignoring)
KLEE: WARNING ONCE: calling external: syscall(16, 0, 21505, 94326022097200) at /home/wei/tmp/klee/runtime/POSIX/fd.c:991 10
KLEE: WARNING ONCE: calling __user_main with extra arguments.
KLEE: WARNING ONCE: Alignment of memory from call "malloc" is not modelled. Using alignment of 8.
KLEE: WARNING ONCE: Alignment of memory from call "calloc" is not modelled. Using alignment of 8.
KLEE: WARNING ONCE: Alignment of memory from call "realloc" is not modelled. Using alignment of 8.

KLEE: WARNING ONCE: calling close_stdout with extra arguments.

<Blank>

KLEE: done: total instructions = 1539344
KLEE: done: completed paths = 47
KLEE: done: generated tests = 47

--- a/tools/klee-replay/klee-replay.c
+++ b/tools/klee-replay/klee-replay.c
@@ -321,9 +321,16 @@ int main(int argc, char** argv) {
     }
   }
 
+  // Executable needs to be converted to an absolute path, as klee-replay calls
+  // chdir just before executing it
+  char executable[PATH_MAX];
+  if (!realpath(argv[optind], executable)) {
+    snprintf(executable, PATH_MAX, "KLEE-REPLAY: ERROR: executable %s:",
+             argv[optind]);
+    perror(executable);
+    exit(1);
+  }
   /* Normal execution path ... */
-
-  char* executable = argv[optind];
 
   /* make sure this process has the CAP_SYS_CHROOT capability, if possible. */
 #ifdef HAVE_SYS_CAPABILITY_H
@@ -336,14 +343,6 @@ int main(int argc, char** argv) {
     fputs("KLEE-REPLAY: ERROR: chroot: root dir should be a parent dir of executable.\n", stderr);
     exit(1);
   }
-
-  /* Verify the executable exists. */
-  FILE *f = fopen(executable, "r");
-  if (!f) {
-    fprintf(stderr, "KLEE-REPLAY: ERROR: executable %s not found.\n", executable);
-    exit(1);
-  }
-  fclose(f);
 
   int idx = 0;
   for (idx = optind + 1; idx != argc; ++idx) {

